pages/instrumenting.md

```diff
@@ -58,12 +58,31 @@ information about the RPC. For instance when calling an HTTP service, providing
 the URI of the call will help with later analysis of requests coming into the
 service.
 
+The primary use case of binary annotations is exact match search. That said, it is
+common to have binary annotations for uncontrolled values, such as exception messages.
+
+**Endpoint**
+Annotations and binary annotations have an endpoint associated with them. With two
+exceptions, this endpoint is associated with the traced process. For example, the
+service name drop-down in the Zipkin UI corresponds with Annotation.endpoint.serviceName
+or BinaryAnnotation.endpoint.serviceName. For the sake of usability, the cardinality
+of Endpoint.serviceName should be bounded. For example, it shouldn't include variables
+or random numbers.
+
+
 **Span**
 
 A set of Annotations and BinaryAnnotations that correspond to a particular RPC.
 Spans contain identifying information such as traceId, spandId, parentId, and
 RPC name.
 
+Spans are usually small. For example, the serialized form is often measured in
+KiB or less. When spans grow beyond orders of KiB, other problems occur, such as
+hitting limits like Kafka message size (1MiB). Even if you can raise message
+limits, large spans will increase the cost and decrease the usability of the
+tracing system. For this reason, be conscious to store data that helps explain
+latency, and don't store data that doesn't.
```

Comment:
Will do

On 12 Sep 2016 00:43, "Yuri Shkuro" <notifications@github.com> wrote:

> In pages/instrumenting.md
> <https://github.com/openzipkin/openzipkin.github.io/pull/35#discussion_r78298285>
> :
>
> >  **Span**
> >
> >  A set of Annotations and BinaryAnnotations that correspond to a particular RPC.
> >  Spans contain identifying information such as traceId, spandId, parentId, and
> >  RPC name.
> >
> > +Spans are usually small. For example, the serialized form is often measured in
> > +KiB or less. When spans grow beyond orders of KiB, other problems occur, such as
> > +hitting limits like Kafka message size (1MiB). Even if you can raise message
> > +limits, large spans will increase the cost and decrease the usability of the
> > +tracing system. For this reason, be conscious to store data that helps explain
> > +latency, and don't store data that doesn't.
>
> I would s/latency/system behavior/ - latency is not the only use case of
> tracing.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/openzipkin/openzipkin.github.io/pull/35/files/eeb91ad188ac5257ec2123b9cf557674c74f3d7c#r78298285>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAD610JBD3mrYPGLBwFkX1Rurz8nR8QGks5qpC-ugaJpZM4IrHCO>
> .
>